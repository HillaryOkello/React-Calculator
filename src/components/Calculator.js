import React, { useState } from 'react';
import Display from './display';
import ButtonPanel from './buttonPanel';
import calculate from '../logic/calculate';
import Navbar from './Navbar';
import './assets/CSS/Calculator.css';

const Calculator = () => {
  const [total, setTotal] = useState(null);
  const [next, setNext] = useState(null);
  const [operation, setOperation] = useState(null);

  const handleClick = (buttonName) => {
    const state = { total, next, operation };

    const newData = calculate(state, buttonName);
    setTotal(newData.total);
    setNext(newData.next);
    setOperation(newData.operation);
  };

  return (
    <>
      <Navbar />
      <div className="container">
        <h2>Let&apos;s do some math!</h2>
        <div className="calculator">
          <Display result={next} />
          <ButtonPanel clickHandler={handleClick} />
        </div>
      </div>
    </>
  );
};
require('react-dom');
window.React2 = require('react');
console.log(window.React1 === window.React2);

export default Calculator;
