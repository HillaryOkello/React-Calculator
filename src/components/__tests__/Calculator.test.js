import React from 'react';
import { Router } from 'react-router-dom';
import { createBrowserHistory } from 'history';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Calculator from '../Calculator';

const history = createBrowserHistory();

it('renders correctly', () => {
  const tree = renderer.create(
    <Router history={history}>
      <Calculator />
    </Router>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('contains Title', () => {
  const { container, getByText } = render(
    <Router history={history}>
      <Calculator />
    </Router>,
  );
  expect(getByText("Let's do some math!")).toBeInTheDocument();
  expect(container.children[1].firstChild).toMatchInlineSnapshot(`
    <h2>
      Let's do some math!
    </h2>
  `);
});

it('contains NavBar', () => {
  const { getByText } = render(
    <Router history={history}>
      <Calculator />
    </Router>,
  );
  expect(getByText('Math Magicians')).toBeInTheDocument();
});

it('contains Display', () => {
  const { container } = render(
    <Router history={history}>
      <Calculator />
    </Router>,
  );
  expect(container.children[1].children[1].firstChild.className).toBe('display');
});

it('contains Panel', () => {
  const { container } = render(
    <Router history={history}>
      <Calculator />
    </Router>,
  );
  expect(container.children[1].children[1].children[1].className).toBe('panel');
});
