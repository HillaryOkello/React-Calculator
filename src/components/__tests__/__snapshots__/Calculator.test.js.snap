// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <div
    className="top"
  >
    <h1>
      Math Magicians
    </h1>
    <div
      className="links"
    >
      <a
        className="link"
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
      <a
        className="link"
        href="/calculator"
        onClick={[Function]}
      >
        Calculator
      </a>
      <a
        className="link"
        href="/quote"
        onClick={[Function]}
      >
        Quote
      </a>
    </div>
  </div>,
  <div
    className="container"
  >
    <h2>
      Let's do some math!
    </h2>
    <div
      className="calculator"
    >
      <p
        className="display"
      >
        0
      </p>
      <div
        className="panel"
      >
        <div
          className="group"
        >
          <button
            onClick={[Function]}
            style={
              Object {
                "background": "orange",
                "font": "3rem 'Syne Mono'",
                "height": "100%",
                "textAlign": "center",
                "width": "25%",
              }
            }
            type="button"
          >
            AC
          </button>
          <button
            onClick={[Function]}
            style={
              Object {
                "background": "orange",
                "font": "3rem 'Syne Mono'",
                "height": "100%",
                "textAlign": "center",
                "width": "25%",
              }
            }
            type="button"
          >
            +/-
          </button>
          <button
            onClick={[Function]}
            style={
              Object {
                "background": "orange",
                "font": "3rem 'Syne Mono'",
                "height": "100%",
                "textAlign": "center",
                "width": "25%",
              }
            }
            type="button"
          >
            %
          </button>
          <button
            onClick={[Function]}
            style={
              Object {
                "background": "orange",
                "font": "3rem 'Syne Mono'",
                "height": "100%",
                "textAlign": "center",
                "width": "25%",
              }
            }
            type="button"
          >
            /
          </button>
        </div>
        <div
          className="group"
        >
          <button
            onClick={[Function]}
            style={
              Object {
                "background": "orange",
                "font": "3rem 'Syne Mono'",
                "height": "100%",
                "textAlign": "center",
                "width": "25%",
              }
            }
            type="button"
          >
            7
          </button>
          <button
            onClick={[Function]}
            style={
              Object {
                "background": "orange",
                "font": "3rem 'Syne Mono'",
                "height": "100%",
                "textAlign": "center",
                "width": "25%",
              }
            }
            type="button"
          >
            8
          </button>
          <button
            onClick={[Function]}
            style={
              Object {
                "background": "orange",
                "font": "3rem 'Syne Mono'",
                "height": "100%",
                "textAlign": "center",
                "width": "25%",
              }
            }
            type="button"
          >
            9
          </button>
          <button
            onClick={[Function]}
            style={
              Object {
                "background": "orange",
                "font": "3rem 'Syne Mono'",
                "height": "100%",
                "textAlign": "center",
                "width": "25%",
              }
            }
            type="button"
          >
            *
          </button>
        </div>
        <div
          className="group"
        >
          <button
            onClick={[Function]}
            style={
              Object {
                "background": "orange",
                "font": "3rem 'Syne Mono'",
                "height": "100%",
                "textAlign": "center",
                "width": "25%",
              }
            }
            type="button"
          >
            4
          </button>
          <button
            onClick={[Function]}
            style={
              Object {
                "background": "orange",
                "font": "3rem 'Syne Mono'",
                "height": "100%",
                "textAlign": "center",
                "width": "25%",
              }
            }
            type="button"
          >
            5
          </button>
          <button
            onClick={[Function]}
            style={
              Object {
                "background": "orange",
                "font": "3rem 'Syne Mono'",
                "height": "100%",
                "textAlign": "center",
                "width": "25%",
              }
            }
            type="button"
          >
            6
          </button>
          <button
            onClick={[Function]}
            style={
              Object {
                "background": "orange",
                "font": "3rem 'Syne Mono'",
                "height": "100%",
                "textAlign": "center",
                "width": "25%",
              }
            }
            type="button"
          >
            -
          </button>
        </div>
        <div
          className="group"
        >
          <button
            onClick={[Function]}
            style={
              Object {
                "background": "orange",
                "font": "3rem 'Syne Mono'",
                "height": "100%",
                "textAlign": "center",
                "width": "25%",
              }
            }
            type="button"
          >
            1
          </button>
          <button
            onClick={[Function]}
            style={
              Object {
                "background": "orange",
                "font": "3rem 'Syne Mono'",
                "height": "100%",
                "textAlign": "center",
                "width": "25%",
              }
            }
            type="button"
          >
            2
          </button>
          <button
            onClick={[Function]}
            style={
              Object {
                "background": "orange",
                "font": "3rem 'Syne Mono'",
                "height": "100%",
                "textAlign": "center",
                "width": "25%",
              }
            }
            type="button"
          >
            3
          </button>
          <button
            onClick={[Function]}
            style={
              Object {
                "background": "orange",
                "font": "3rem 'Syne Mono'",
                "height": "100%",
                "textAlign": "center",
                "width": "25%",
              }
            }
            type="button"
          >
            +
          </button>
        </div>
        <div
          className="group"
        >
          <button
            onClick={[Function]}
            style={
              Object {
                "background": "orange",
                "font": "3rem 'Syne Mono'",
                "height": "100%",
                "textAlign": "center",
                "width": "25%",
              }
            }
            type="button"
          >
            0
          </button>
          <button
            onClick={[Function]}
            style={
              Object {
                "background": "orange",
                "font": "3rem 'Syne Mono'",
                "height": "100%",
                "textAlign": "center",
                "width": "25%",
              }
            }
            type="button"
          >
            .
          </button>
          <button
            onClick={[Function]}
            style={
              Object {
                "background": "orange",
                "font": "3rem 'Syne Mono'",
                "height": "100%",
                "textAlign": "center",
                "width": "25%",
              }
            }
            type="button"
          >
            =
          </button>
        </div>
      </div>
    </div>
  </div>,
]
`;
